docs/1-Getting-Started/4-How-it-works.md

```diff
@@ -34,56 +33,54 @@ scripts:
     - kool run artisan key:generate
 ```
 
-Usage:
+Using the `artisan` command:
+
 ```bash
 kool run artisan tinker
 ```
 
-You can pass in after `kool run my-script` any other arguments you wish to pass down the encapsulated command.
-
-> The arguments passing at this moment is limited to single-line scripts only.
+#### Types of Commands
 
-#### Arguments to kool run <script>
+The **kool.yml** file is not just for **kool** commands. You can add any type of command you usually run in your shell, such as `cat`, `cp`, `mv`, etc.
 
-Single commands like **artisan** are kind of aliases, so anything you input will be forwarded to the actual command, so if you run: **kool run artisan key:generate** it will basically translate into: **kool exec app php artisan key:generate**.
+However, there is one important caveat - the commands you add to **kool.yml** are parsed and executed by the **kool** binary, and not in a general **bash** context. This means you **cannot** directly use **bash** control structures like `if []; then fi`, or redirection with pipes (`cmd | cmd2`). With that said, **we do support** input and output redirection (see below). W00t!
 
-Multiple commands like **setup** will not forward your input, so **kool run setup something** will run every script and **something** will be ignored.
+> If you need to add a more complex shell command, you can use something like `kool docker <some bash image> bash -c ""`, which will parse any bash script you need.
 
-#### What kind of commands can be encasulated on `kool.yml`
+#### Adding Arguments
 
-This is not meant only for `kool` commands, you can add any type commands as you usually run them in your shell like `cat`, `cp`, `mv`, etc.
+At the end of single line scripts like `kool run <script-name>`, you can also add arguments you want to pass down to the encapsulated command. Single line commands, such as `artisan`, are like aliases, whereby additional arguments are forwarded to the actual command. For example, `kool run artisan key:generate` basically becomes `kool exec app php artisan key:generate`.
 
-There is just one caveat we need to be aware of - the commands within a script on `kool.yml` are parsed and executed by `kool` and not in a general `bash` context, so you **cannot** directly use bash script structures like `if []; then fi` or redirection with pipes (`cmd | cmd2`). Although, we support output redirection (see below).
+At this time, adding arguments is only supported by **single line** scripts. **Multi-line** commands like `setup` will **not** forward your arguments. This means `kool run setup something` will run each script in sequence, and completely ignore the `something` argument added to the end.
```

Comment:
Actually the multi-line commands will error out, and not simply ignore the argument. We need to indicate that.

Reply:
@fabriciojs - how about you merge my PR upstream into yours, and then I make this edit directly via the GH UI?